### PR TITLE
[WIP] Split api/common/server impls

### PR DESCRIPTION
**Pull request by clienthax**

### CODE REVIEW

File: api/src/main/java/me/confuser/banmanager/api/Entity.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+package me.confuser.banmanager.api;
+
+import com.sun.istack.internal.Nullable;
+import lombok.NonNull;
+
+import java.util.UUID;
```
Comment (confuser) [inline api/src/main/java/me/confuser/banmanager/api/Entity.java]: @lucko do you have any of these available in a shareable lib by any chance? Looking at adding Sponge support on top of Bukkit and LuckPerms seems to have a great way of doing this

File: api/src/main/java/me/confuser/banmanager/api/event/EventBus.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    /**
+     * Gets a set of all registered handlers for a given event.
+     *
+     * @param eventClass the event to find handlers for
+     * @param <T>        the event class
+     * @return an immutable set of event handlers
+     */
+    @NonNull <T extends BanManagerEvent> Set<EventHandler<T>> getHandlers(@NonNull Class<T> eventClass);
+
+}
```
Comment (confuser) [inline api/src/main/java/me/confuser/banmanager/api/event/EventBus.java]: @clienthax what's this used for? Proxy support?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+defaultTasks 'licenseFormat', 'build'
+
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'maven'
+    //apply plugin: 'net.minecrell.licenser'
+	//apply plugin: "io.freefair.lombok"
+
+    group = 'me.confuser.banmanager'
+    version = '6.0.1-SNAPSHOT'
```
Comment (confuser) [inline build.gradle]: Can we set this to `7.0.0-SNAPSHOT` please? They'll be some major breaking changes with this which will break compatibility with other plugins.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+    }
+
+    jar {
+        from '../LICENSE.txt'
+    }
+
+    def determinePatchVersion = {
```
Comment (confuser) [inline build.gradle]: Not sure this is necessary

File: bukkit/.gitignore [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+.gradle
+tmp
+
+# Package Files #
+*.jar
+*.war
+*.ear
+
+# Web Interface Files #
+banmanagement/settings.php
```
Comment (confuser) [inline bukkit/.gitignore]: These can be removed, legacy from when the web ui was in the same repo

File: bukkit/build.gradle [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    compileOnly 'org.spigotmc:spigot-api:1.14-R0.1-SNAPSHOT'
+    compileOnly 'net.kyori:text-adapter-bukkit:3.0.2'
+    compileOnly 'me.lucko:commodore:1.3'
+    compileOnly('net.milkbowl.vault:VaultAPI:1.6') {
+        exclude(module: 'bukkit')
+    }
+    compileOnly 'lilypad.client.connect:api:0.0.1-SNAPSHOT'
+
+    // migration plugins
+    compileOnly 'org.tyrannyofheaven.bukkit:zPermissions:1.3'
```
Comment (confuser) [inline bukkit/build.gradle]: Not needed

File: bukkit/build.gradle [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    maven { url 'https://libraries.minecraft.net/' }
+}
+
+dependencies {
+    compile project(':common')
+
+    compileOnly 'org.spigotmc:spigot-api:1.14-R0.1-SNAPSHOT'
+    compileOnly 'net.kyori:text-adapter-bukkit:3.0.2'
+    compileOnly 'me.lucko:commodore:1.3'
+    compileOnly('net.milkbowl.vault:VaultAPI:1.6') {
```
Comment (confuser) [inline bukkit/build.gradle]: Not needed

File: bukkit/build.gradle [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    maven { url 'https://hub.spigotmc.org/nexus/content/repositories/snapshots/' }
+    maven { url 'https://libraries.minecraft.net/' }
+}
+
+dependencies {
+    compile project(':common')
+
+    compileOnly 'org.spigotmc:spigot-api:1.14-R0.1-SNAPSHOT'
+    compileOnly 'net.kyori:text-adapter-bukkit:3.0.2'
+    compileOnly 'me.lucko:commodore:1.3'
```
Comment (confuser) [inline bukkit/build.gradle]: Not sure if this is needed? Seems to be for auto tab completions

File: bukkit/build.gradle [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    compileOnly('net.milkbowl.vault:VaultAPI:1.6') {
+        exclude(module: 'bukkit')
+    }
+    compileOnly 'lilypad.client.connect:api:0.0.1-SNAPSHOT'
```
Comment (confuser) [inline bukkit/build.gradle]: Not needed unless we add direct support for proxies (event driven vs polling for data changes)

File: bukkit/build.gradle [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        exclude(module: 'AccountsClient')
+    }
+    compileOnly 'com.github.gustav9797:PowerfulPermsAPI:4.5.2'
+    compileOnly 'org.anjocaido:GroupManager:1.4'
+    compileOnly 'de.bananaco:bpermissions-api:2.12'
+    compileOnly('com.platymuus:bukkit-permissions:2.5') {
+        exclude(module: 'bukkit')
+        exclude(module: 'metrics')
+    }
+
```
Comment (confuser) [inline bukkit/build.gradle]: None of the permission dependencies needed

File: common/src/main/java/me/confuser/banmanager/BmAPI.java [hunk 11]
```diff
@@ -35,17 +35,6 @@ public static PlayerData getPlayer(UUID uuid) throws SQLException {
...
   }
 
-  /**
-   * @param player
-   *
-   * @return PlayerData
-   *
-   * @throws SQLException
-   */
-  public static PlayerData getPlayer(Player player) throws SQLException {
```
Comment (confuser) [inline common/src/main/java/me/confuser/banmanager/BmAPI.java]: @clienthax what's the reason for removing these?

Comment (clienthax) [reply common/src/main/java/me/confuser/banmanager/BmAPI.java]: Bukkit Player class, can't have it in the API as its implementation specific.
If we wan't to keep it we could delegate down to the other handlers, but it seems easier to just scrap it off completly.

File: bukkit/build.gradle (hunk 10 above)
Comment (clienthax) [reply bukkit/build.gradle]: Yeah a lot of the build gradle parts can be cleaned up, i basically just copied luckperms for the time being

**Comment by confuser** [comment]
Superseded by #795

**Pull request closed without merging**